library/init/meta/tactic.lean

```diff
@@ -413,11 +414,20 @@ meta constant abstract_hash : expr → tactic nat
 meta constant abstract_weight : expr → tactic nat
 meta constant abstract_eq     : expr → expr → tactic bool
 /- (induction_core m H rec ns) induction on H using recursor rec, names for the new hypotheses
-   are retrieved from ns. If ns does not have sufficient names, then use the internal binder names in the recursor. -/
-meta constant induction_core : transparency → expr → name → list name → tactic unit
+   are retrieved from ns. If ns does not have sufficient names, then use the internal binder names
+   in the recursor.
+   It returns for each new goal a list of new hypotheses and a substitution mapping for hypothesis
+   depending on H. The names in the substitution mapping are only valid in the original goal, not in
+   the type contexts of the new goals. -/
+meta constant induction_core : transparency → expr → name → list name → tactic (list (list expr × list (name × expr)))
```

Comment:
This is cool, you are also returning the hypotheses that have been renamed.
BTW, we are not very precise what you mean by "renaming" here and in the rest of the API.
Each hypothesis has two names: internal and user. The user names are for user interaction (e.g., pretty printing). The internal names are unique. 
This tactic does not rename the "user" names, only the internal ones.
Any ideas on how to explain this stuff in the API?